pkg/ebpftracer/tracer.go

```diff
@@ -195,45 +197,16 @@ func (t *Tracer) GetEventName(id events.ID) string {
 	return ""
 }
 
-func (t *Tracer) signalReadLoop(ctx context.Context) error {
-	eventsReader, err := perf.NewReader(t.module.objects.Signals, t.cfg.EventsPerCPUBuffer)
-	if err != nil {
-		return err
-	}
-	defer eventsReader.Close()
-
-	for {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
-		record, err := eventsReader.Read()
-		if err != nil {
-			if t.cfg.DebugEnabled {
-				t.log.Warnf("reading signals: %v", err)
-			}
-			continue
-		}
-		if record.LostSamples > 0 {
-			t.log.Warnf("lost %d signals", record.LostSamples)
-			metrics.AgentKernelLostEventsTotal.Add(float64(record.LostSamples))
-			continue
-		}
-		metrics.AgentPulledEventsTotal.Inc()
-
-		if err := t.decodeAndHandleSignal(ctx, record.RawSample); err != nil {
-			if t.cfg.DebugEnabled || errors.Is(err, ErrPanic) {
-				t.log.Errorf("decoding signal: %v", err)
-			}
-			metrics.AgentDecodeEventErrorsTotal.Inc()
-			continue
-		}
-	}
+func (t *Tracer) signalEventsReadLoop(ctx context.Context) error {
+	return t.runPerfBufReaderLoop(ctx, t.module.objects.Events)
```

Comment:
Looks like it should read from SignaEvents map.

Reply:
How on earth did this ever work 🤔 
All process events clearly write to the signal perf buf ??? Either something is severely broken with my local setup, or  I am missing something.

Reply:
Well, another mistake covered for this one. I never write to the signal buf
https://github.com/castai/kvisor/pull/291/files#diff-44b3eab9a3ef27a5ad8543372ffafbec61c5bb1ae491920f402589d270dd27cdR509